packages/tangle/scheduler.go

```diff
@@ -178,6 +186,39 @@ func (s *Scheduler) NodeQueueSizes() map[identity.ID]int {
 	return nodeQueueSizes
 }
 
+// MaxBufferSize returns the max size of the buffer.
+func (s *Scheduler) MaxBufferSize() int {
+	s.mu.Lock()
+	defer s.mu.Unlock()
```

Comment:
Is there a specific reason why we use a lock rather than a RW lock?

Reply:
That's a good question. I just used whatever was already in place, but we could change it to RW lock. I'll do it :)